spec/rubocop/cop/lint/useless_assignment_spec.rb

```diff
@@ -1326,6 +1326,26 @@ def some_method(name: value, **)
         do_something { _1 == var }
       RUBY
     end
+
+    it 'does not register an offense when the variable is assigned and later used' do
+      expect_no_offenses(<<~RUBY)
+        var = nil
+
+        do_something { var = _1 }
+
+        something_else(var)
+      RUBY
+    end
+
+    it 'registers an offense when the variable is assigned and but unused' do
+      expect_offense(<<~RUBY)
+        var = nil
+        ^^^ Useless assignment to variable - `var`.
+
+        do_something { var = _1 }
+                       ^^^ Useless assignment to variable - `var`.
+      RUBY
+    end
```

Comment:
Hmm... I just added this test as the inverse of the one above, but it looks like it fails. I think it's because
- `var = nil` is considered fine, because a block later uses it
- `var = _1` is considered fine, because it's assigning to a shadowed variable

I don't think there's anything checking that the variable is used in neither the block, nor the surrounding scope.

I'm not sure how to go about implementing that, so I may just remove the test and accept the unlikely false negative. 🤷

Reply:
Yep. It's an existing issue that also happens with normal block (not numbered block). So, can you remove this test?